src/features/embalm/stepContent/components/Bundlr.tsx

```diff
@@ -24,31 +14,26 @@ export function Bundlr({ children }: { children?: React.ReactNode }) {
   const { connectToBundlr, disconnectFromBundlr, isConnected } = useBundlrSession();
   const { formattedBalance } = useGetBalance();
   const { uploadPrice, formattedUploadPrice } = useUploadPrice();
-  const [inputAmount, setInputAmount] = useState(
-    uploadPrice.eq(ethers.constants.Zero)
-      ? undefined
-      : parseFloat(ethers.utils.formatUnits(uploadPrice)).toFixed(uploadPriceDecimals)
-  );
+
   const { balanceOffset } = useSelector(x => x.bundlrState);
   const { chain } = useNetwork();
 
-  const isInputAmountValid = !!inputAmount && parseFloat(inputAmount) > 0;
-
-  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
-    const { value } = event.target;
-    if (Number(value) < 0 || !ethers.utils.parseEther(value)) return;
-    const inputLimit = 46;
-    setInputAmount(value.slice(0, inputLimit));
-  }
-
   function handleDisconnect() {
     disconnectFromBundlr();
   }
 
+  const { BundlrInput, inputAmountBN } = useBundlrInput({
+    initialAmount:
+      uploadPrice && !uploadPrice.eq(ethers.constants.Zero)
+        ? parseFloat(ethers.utils.formatUnits(uploadPrice))
+        : undefined,
+    inputHeight: '40px',
```

Comment:
With the change suggested above we can remove the `inputHeight` and include it in the `<BundlrInput />` component below.